### PR TITLE
 [TM] Add spec for DialogManagerAndroid 

### PR DESCRIPTION
**Pull request by uqmessias**
## Summary

Part of #24875.

## Changelog

[General] [Added] - TM add spec for DialogManagerAndroid

## Test Plan

Flow green.

### CODE REVIEW

File: Libraries/NativeModules/specs/NativeDialogManagerAndroid.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  +cancelable?: boolean,
+|};
+
+export interface Spec extends TurboModule {
+  +getConstants: () => {|
+    buttonClicked: DialogAction,
+    dismissed: DialogAction,
+    buttonPositive: DialogButtonKey,
+    buttonNegative: DialogButtonKey,
+    buttonNeutral: DialogButtonKey,
```
Comment (RSNara) [inline Libraries/NativeModules/specs/NativeDialogManagerAndroid.js]: Looking through the implementation of `DialogManagerAndroid`, it looks like `buttonPositive`, `buttonNegative`, and `buttonNeutral` are numbers? 

#### DialogModule.java
```Java
  /* package */ static final String ACTION_BUTTON_CLICKED = "buttonClicked";
  /* package */ static final String ACTION_DISMISSED = "dismissed";
  /* package */ static final String KEY_TITLE = "title";
  /* package */ static final String KEY_MESSAGE = "message";
  /* package */ static final String KEY_BUTTON_POSITIVE = "buttonPositive";
  /* package */ static final String KEY_BUTTON_NEGATIVE = "buttonNegative";
  /* package */ static final String KEY_BUTTON_NEUTRAL = "buttonNeutral";
  /* package */ static final String KEY_ITEMS = "items";
  /* package */ static final String KEY_CANCELABLE = "cancelable";

  /* package */ static final Map<String, Object> CONSTANTS = MapBuilder.<String, Object>of(
      ACTION_BUTTON_CLICKED, ACTION_BUTTON_CLICKED,
      ACTION_DISMISSED, ACTION_DISMISSED,
      KEY_BUTTON_POSITIVE, DialogInterface.BUTTON_POSITIVE,
      KEY_BUTTON_NEGATIVE, DialogInterface.BUTTON_NEGATIVE,
      KEY_BUTTON_NEUTRAL, DialogInterface.BUTTON_NEUTRAL);
```

#### DialogInterface.java
```Java
public interface DialogInterface {
    /** The identifier for the positive button. */
    int BUTTON_POSITIVE = -1;

    /** The identifier for the negative button. */
    int BUTTON_NEGATIVE = -2;

    /** The identifier for the neutral button. */
    int BUTTON_NEUTRAL = -3;
```

File: Libraries/NativeModules/specs/NativeDialogManagerAndroid.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  +buttonNegative?: string,
+  +buttonNeutral?: string,
+  +items?: Array<string>,
+  +cancelable?: boolean,
+|};
+
+export interface Spec extends TurboModule {
+  +getConstants: () => {|
+    buttonClicked: DialogAction,
+    dismissed: DialogAction,
```
Comment (RSNara) [inline Libraries/NativeModules/specs/NativeDialogManagerAndroid.js]: Unfortunately, the Codegen system doesn't support unions. 😔

 So, I think it's best if we change this to `string`.

File: Libraries/NativeModules/specs/NativeDialogManagerAndroid.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    buttonPositive: DialogButtonKey,
+    buttonNegative: DialogButtonKey,
+    buttonNeutral: DialogButtonKey,
+  |};
+  +showAlert: (
+    config: DialogOptions,
+    onError: (string) => void,
+    onAction: (action: DialogAction, buttonKey: DialogButtonKey) => void,
```
Comment (RSNara) [inline Libraries/NativeModules/specs/NativeDialogManagerAndroid.js]: Looking through the code, it seems like `onAction` can also be invoked without the `buttonKey`:

```Java
    @Override
    public void onDismiss(DialogInterface dialog) {
      if (!mCallbackConsumed) {
        if (getReactApplicationContext().hasActiveCatalystInstance()) {
          mCallback.invoke(ACTION_DISMISSED);
          mCallbackConsumed = true;
        }
      }
    }
```

So we should make this argument nullable.

File: Libraries/NativeModules/specs/NativeDialogManagerAndroid.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+'use strict';
+
+import type {TurboModule} from 'RCTExport';
+import * as TurboModuleRegistry from 'TurboModuleRegistry';
+
+import Platform from '../../Utilities/Platform';
+
+type DialogAction = 'buttonClicked' | 'dismissed';
+type DialogButtonKey = 'buttonPositive' | 'buttonNegative' | 'buttonNeutral';
```
Comment (uqmessias) [inline Libraries/NativeModules/specs/NativeDialogManagerAndroid.js]: @RSNara, I saw one of your comments in another PR about the codegen not supporting enum or union types. So, based on that, these two types (`DialogAction` and `DialogButtonKey`) are not supported by the codegen, right?

File: Libraries/NativeModules/specs/NativeDialogManagerAndroid.js (hunk 2 above)
Comment (uqmessias) [reply Libraries/NativeModules/specs/NativeDialogManagerAndroid.js]: What about keeping this as it is and adding and updating `DialogAction` to `type DialogAction = string`?

File: Libraries/NativeModules/specs/NativeDialogManagerAndroid.js (hunk 1 above)
Comment (uqmessias) [reply Libraries/NativeModules/specs/NativeDialogManagerAndroid.js]: @RSNara,
Thank you for reviewing my PR. I already made the required changes.

Can you take a look on it again, pls?

**Comment by uqmessias** [comment]
@RSNara,
Can you take a look on this again, pls?

**Pull request closed without merging**